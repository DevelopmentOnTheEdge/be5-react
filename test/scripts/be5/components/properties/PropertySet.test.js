import React from 'react';
import PropertySet from '../../../../../src/scripts/be5/components/properties/propertySet';
import renderer from 'react-test-renderer';
import {shallow, mount, render} from 'enzyme';

it('renders without crashing', () => {
	const testJson = require('./testJson.json');
	const handle = jest.fn();

	mount(<PropertySet bean={testJson} onChange={handle}/>);
});

it('renders without crashing readOnly', () => {
	const bean = require('./testJson.json');
	for(let item in bean.meta) {
    bean.meta[item]['readOnly'] = true;
	}

	const handle = jest.fn();

	mount(<PropertySet bean={bean} onChange={handle}/>);
});

it('simple property set', () => {
	const bean = {
		 "values": {
			 "number": "",
		 },
		 "meta": {
			 "/number": {},
		 },
		 "order": [
			 "/number",
		 ]
	};

	const component = renderer.create(
    <PropertySet bean={bean} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('empty property set', () => {
	const bean = {
		 "values": {},
		 "meta": {},
		 "order": []
	};

	const component = renderer.create(
    <PropertySet bean={bean} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
