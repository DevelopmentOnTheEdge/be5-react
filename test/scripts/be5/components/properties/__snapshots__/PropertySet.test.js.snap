// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty property set 1`] = `
<div
  className="property-set row"
/>
`;

exports[`simple property set 1`] = `
<div
  className="property-set row"
>
  <div
    className="form-group property col-xs-12 required"
  >
    <label
      className="form-control-label"
      htmlFor="numberField"
    >
      numberField
    </label>
    <div
      className="controls"
    >
      <input
        className="form-control"
        disabled={undefined}
        id="numberField"
        onChange={[Function]}
        placeholder={undefined}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
